cmd/zdb/zdb.c

```diff
@@ -6489,10 +6530,9 @@ zdb_read_block(char *thing, spa_t *spa)
 		 */
 		zio_nowait(zio_vdev_child_io(zio, bp, vd, offset, pabd,
 		    psize, ZIO_TYPE_READ, ZIO_PRIORITY_SYNC_READ,
-		    ZIO_FLAG_DONT_CACHE | ZIO_FLAG_DONT_QUEUE |
```

Comment:
Why was `ZIO_FLAG_DONT_QUEUE` removed from here?  Did it result in problems?  I'm not certain, since it wasn't commented, but my hunch is it was included to ensure the vdev queue never tries to create an aggregate IO for this read via `vdev_queue_aggregate()`.

Reply:
This was to get around this issue:
```
# zdb -R testpool 1.0:8038600:600:d
Found vdev: /root/x.2
tree->avl_numnodes > 0
ASSERT at ../../module/avl/avl.c:747:avl_remove()Aborted
```
I'd have to go back and look at the reason that the avl_tree was already empty when the zio was not queued..  I'll let you know.

Reply:
ZIO_FLAG_DONT_QUEUE is only set in zdb (before this PR), in vdev_queue_aggregate, and in vdev_queue_io.  It looks like it is only used to prevent re-queueing of a queued zio (vdev_queue.c line 808).  So I don't believe zdb should be setting this flag, but would like confirmation.

Reply:
After looking at this again, you're right the makes sense.  Though I still personally find the name a bit confusing!  OK, so this does need to go.